src/app/tabs/compileTab/compilerContainer.js

```diff
@@ -289,11 +326,54 @@ class CompilerContainer {
            (version.includes('nightly') && this._view.includeNightlies.checked)
   }
 
-  _updateVersionSelector () {
+  /**
+   * Update & load compiler
+   * @param {Object} pragma - Pragma version information
+   * @param {string} pragma.start - Version of pragma, Start in case of range
+   * @param {boolean} [pragma.startInclusive] - true, if start version is inclusive in range
+   * @param {string} [pragma.end] - End version of pragma in case of range
+   * @param {boolean} [pragma.endInclusive] - true, if end version is inclusive in range
+   * @param {boolean} [pragma.isFixed] - true, if pragma is not a range
+   */
+  _updateVersionSelector (pragma = null) {
     // update selectedversion of previous one got filtered out
     if (!this._shouldBeAdded(this.data.selectedVersion)) {
       this.data.selectedVersion = this.data.defaultVersion
     }
+    if (pragma && pragma.start) {
+      const allversions = this.data.allversions.filter(build => (!build.prerelease))
+      const startIndex = allversions.findIndex(build => build.version === pragma.start)
+      if (startIndex > -1) {
+        let selectedVersion
+        if (pragma.isFixed) {
+          selectedVersion = allversions[startIndex].path
+        } else {
+          let maxIndex = 0
+          if (pragma.end) {
+            let endIndex = allversions.findIndex(build => build.version === pragma.end)
+            if (endIndex > -1) {
+              if (!pragma.endInclusive) {
+                endIndex += 1
+              }
+              if (startIndex === endIndex && !pragma.startInclusive) {
+                endIndex = -1
+              }
+              maxIndex = endIndex
+            }
+          }
+          if (maxIndex > -1) {
+            selectedVersion = allversions[maxIndex].path
+          }
+        }
+        if (this.data.selectedVersion === selectedVersion) {
```

Comment:
maybe it's not selected version by this logic but also fine. for example for a range from 0.3.0 to 0.6.0 you can pick 0.5.12 but 0.5.0 is also fine.

Reply:
selected version is here something which will be selected from compiler dropdown